### PR TITLE
Added Auto LF and .editorconfig settings for aligning formatting rules

### PR DESCRIPTION
**Pull request by oskardudycz**
@jeremydmiller @mysticmind @jokokko if you're fine with this `.editorsettings` then I'll do code clean up and change warnings settings into errors.

See rules explanation: https://docs.microsoft.com/en-US/visualstudio/ide/editorconfig-code-style-settings-reference?view=vs-2017

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,6 @@
+* text=auto *.sh eol=lf
```
Comment (mysticmind) [inline .gitattributes]: Please move `*.sh eol=lf` to a new line

**Comment by jeremydmiller** [comment]
I don't see anything in there that would bug me. Might also be worthwhile to add R# settings too though

**Comment by jeremydmiller** [comment]
I don't see anything in there that would bug me. Might also be worthwhile to add R# settings too though

**Comment by oskardudycz** [comment]
@jeremydmiller I'm not using currently Resharper, but reading that it appears that Resharper should respect `.editorconfig`

**Comment by oskardudycz** [comment]
@jokokko imho it's better to keep it in new line, for me it's not readable and confusing when we have it in single line.

**Pull request merged**